model/src/application/command/option.rs

```diff
@@ -179,6 +181,18 @@ impl Serialize for CommandOption {
                 required: data.required,
                 kind: self.kind(),
             },
+            CommandOption::Unknown(_) => CommandOptionEnvelope {
+                autocomplete: false,
+                channel_types: None,
+                choices: None,
+                description: "",
+                max_value: None,
+                min_value: None,
+                name: "",
+                options: None,
+                required: false,
+                kind: self.kind()
+            }
```

Comment:
This is unfortunate. Although my experience is only tangentially related, after usage of our interactions types I sometimes feel they're far more verbose by having enums like this. Makes me wonder if it'd be simpler for everyone involved to give types like this the ol' Channel treatment.

Reply:
Flattening would surely make these simpler, i think we best leave it as this for this PR and look at such things in a separate PR later